.github/workflows/test.yml

```diff
@@ -130,7 +136,7 @@ jobs:
         run: echo "OPENSSL_CONF=$(pwd)/test/openssl/fixtures/ssl/openssl_fips.cnf" >> $GITHUB_ENV
         if: matrix.fips-enabled
 
-      - name: set fips enviornment variable for testing.
+      - name: set fips environment variable for testing.
```

Comment:
Thank you for finding and fixing a typo. However, I don't think this is related to this PR's main topic. Maybe better to send another PR for that.

Reply:
I opened the PR https://github.com/ruby/openssl/pull/641 for that.

Reply:
I already merged the commit only fixing the typo. And I think after you will finish the review and fix everything in this PR, I think you (or a maintainer) needs to squash this PR on the latest master branch. We call it rebasing the PR on master branch. Could you do that?

Reply:
The typo fix came from my commit, which modified the Actions workflow file, which isn't part of the gem, or copied to ruby/ruby, etc.

Fixing a pair of transposed comment characters in a file that is already being modified seemed trivial...

Reply:
If you squash the commits in this PR to the 1 commit, it's nice to have the following line in the bottom of the commit message. You can check the examples in the past commits by `git log` on the `master` branch. 

```
Co-authored-by: MSP-Greg <Greg.mpls@gmail.com>
```

Reply:
It's often nice to see changes as the PR 'evolves'.  Maintainers can always use the 'Squash and Merge' option.

![Squash_and_Merge_sm](https://github.com/ruby/openssl/assets/15081176/8f049c0b-54af-4431-9f67-f9405ed76df1)